content/en/blog/_posts/2022-10-27-live-and-let-live-with-kluctl-and-ssa.md

```diff
@@ -0,0 +1,174 @@
+(Helm/Kustomize/...) in a manageable and unified way."
+
+To get a basic understanding of Kluctl, I suggest to visit the [kluctl.io](https://kluctlio)
+website and read through the documentation and tutorials, for example the 
+[microservices demo tutorial](https://kluctl.io/docs/guides/tutorials/microservices-demo/).
+As an alternative, you can watch [Hands-on Introduction to kluctl](https://www.youtube.com/watch?v=9LoYLjDjOdg)
+from the Rawkode Academy YouTube channel which shows a hands-on demo session.
+
+There is also a [Kluctl delivery scenario](https://github.com/podtato-head/podtato-head/tree/main/delivery/kluctl)
+available in the [podtato-head](https://github.com/podtato-head/podtato-head) demo project.
+
+## Live and let live
+
+One of the main philosophies that Kluctl follows is ["live and let live"](https://kluctl.io/docs/philosophy/#live-and-let-live),
+meaning that it will try its best to work in conjunction with any other tool or
+controller running outside or inside your clusters. Kluctl will not overwrite
+any fields that it lost ownership of, unless you explicitly tell it to do so.
+
+Achieving this would not have been possible (or at least be several magnitudes
+harder) without the use of Server-side apply. Server-side apply allows Kluctl
+to detect when ownership for a field got lost, for example when another controller
+or operator updates that field to another value. Kluctl can then decide on a 
+field-by-field basis if a force-apply is required and then retry based on these
+decisions.
+
+## The days before Server-side apply
+
+The first versions of Kluctl were based on shelling out to `kubectl` and thus
+implicitly relied on client-side apply. At that time, Server-side apply was
+still alpha and quite buggy. And to be honest, I didn't even know it was a
+thing at that time.
+
+The way client-side apply worked had some serious drawbacks. The most obvious one
+(it was guaranteed that you'd stumble on this by yourself if enough time passed)
+is that it relied on an annotation (kubectl.kubernetes.io/last-applied-configuration)
+being added to the object, bringing in all the limitations and issues with huge
+annotation values. A good example of such issues are
+[CRDs being so large](https://github.com/prometheus-operator/prometheus-operator/issues/4439),
+that they don't fit into the annotation's value anymore.
+
+Another drawback can be seen just by looking at the name (**client**-side apply).
+Being **client** side means that each client has to provide the apply-logic on
+its own, which at that time was only properly implemented inside `kubectl`,
+making it hard to be replicated inside controllers.
+
+This added `kubectl` as a dependency (either as an executable or in the form of
+Go packages) to all controllers that wanted to leverage the apply-logic.
+
+However, even if one managed to get client-side apply running from inside a
+controller, you ended up with a solution that gave no control about how it
```

Comment:
IMO, the clause `you ended up with a solution that gave no control about how it worked internally` should be something like
`you ended up with a solution that gave no clues about how it worked internally`
or
`you ended up with a solution that gave no control over how it worked internally`

Reply:
I'd go with the latter suggestion.